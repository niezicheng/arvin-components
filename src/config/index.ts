// 类型
export enum EType {
  default = 'default',
  primary = 'primary',
  dashed = 'dashed',
  link = 'link',
  text = 'text',
}
export type TType = keyof typeof EType;

// 尺寸
export enum ESize {
  small = 'small',
  middle = 'middle',
  large = 'large',
}
export type TSize = keyof typeof ESize;

// 形状
export enum EShape {
  default = 'default',
  round = 'round',
  circle = 'circle',
  square = 'square',
}
export type TShape = keyof typeof EShape;

// 状态
export enum EStatus {
  info = 'info',
  success = 'success',
  warning = 'warning',
  error = 'error',
}
export type TStatus = keyof typeof EStatus;

// 方向
export enum EDirection {
  vertical = 'vertical',
  horizontal = 'horizontal',
}
export type TDirection = keyof typeof EDirection;

// 对其方式
export enum EAlign {
  start = 'start',
  end = 'end',
  center = 'center',
  baseline = 'baseline',
}
export type TAlign = keyof typeof EAlign;
