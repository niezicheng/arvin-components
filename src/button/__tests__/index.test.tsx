import { Button } from '@arvin/react-ui';
import { render } from '@testing-library/react';
import React from 'react';
import { TSize, TStatus, TType } from '../../config';
import { ButtonShape, ButtonSize, ButtonStatus, ButtonType } from '../demo';

describe('<Button />', () => {
  test('should render default', () => {
    const { container } = render(<Button>default</Button>);
    expect(container).toMatchSnapshot();
  });

  test('should render button with type', () => {
    const types: Array<TType> = [
      'primary',
      'default',
      'dashed',
      'text',
      'link',
    ];
    const { container } = render(<ButtonType />);
    container.querySelectorAll('.ar-button').forEach((ele, index) => {
      expect(ele).toHaveClass(`ar-button--${types[index]}`);
    });
  });

  test('should render button with shape', () => {
    const shapes = ['circle', 'round'];
    const { container } = render(<ButtonShape />);
    container.querySelectorAll('.ar-button').forEach((ele, index) => {
      expect(ele).toHaveClass(`ar-button--${shapes[index]}`);
    });
  });

  test('should render button with size', () => {
    const sizes: Array<TSize> = ['small', 'middle', 'large'];
    const { container } = render(<ButtonSize />);
    container.querySelectorAll('.ar-button').forEach((ele, index) => {
      expect(ele).toHaveClass(`ar-button--${sizes[index]}`);
    });
  });

  test('should render button with status', () => {
    const statuses: Array<TStatus> = ['info', 'success', 'warning', 'error'];
    const { container } = render(<ButtonStatus />);
    container.querySelectorAll('.ar-button').forEach((ele, index) => {
      expect(ele).toHaveClass(`ar-button--${statuses[index]}`);
    });
  });

  test('should render button with loading', () => {
    const { getByText } = render(
      <>
        <Button loading={true}>loading</Button>
        <Button loading={false}>not loading</Button>
      </>,
    );
    expect(getByText('loading')).toMatchSnapshot();
    expect(getByText('not loading')).toMatchSnapshot();
  });

  test('should render button with disabled', () => {
    const { getByText } = render(
      <>
        <Button disabled={true}>disabled</Button>
        <Button disabled={false}>not disabled</Button>
      </>,
    );
    expect(getByText('disabled')).toMatchSnapshot();
    expect(getByText('not disabled')).toMatchSnapshot();
  });

  test('should render button with children', () => {
    const { getByText } = render(
      <>
        <Button>children</Button>
      </>,
    );
    expect(getByText('children')).toMatchSnapshot();
  });

  test('should render button with className', () => {
    const { getByText } = render(
      <>
        <Button className="test">className</Button>
      </>,
    );
    expect(getByText('className')).toMatchSnapshot();
  });

  test('should render button with style', () => {
    const { getByText } = render(
      <>
        <Button style={{ color: 'red' }}>style</Button>
      </>,
    );
    expect(getByText('style')).toMatchSnapshot();
  });

  test('should render button with onClick', () => {
    const onClick = jest.fn();
    const { getByText } = render(
      <>
        <Button onClick={onClick}>onClick</Button>
      </>,
    );
    expect(getByText('onClick')).toMatchSnapshot();
    getByText('onClick').click();
    expect(onClick).toBeCalled();
  });
});
