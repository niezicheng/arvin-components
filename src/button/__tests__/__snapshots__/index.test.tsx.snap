// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render button with children 1`] = `
<button
  class="ar-button ar-button--default ar-button-- ar-button--middle ar-button--loading"
  type="button"
>
  children
</button>
`;

exports[`<Button /> should render button with className 1`] = `
<button
  class="ar-button ar-button--default ar-button-- ar-button--middle ar-button--loading test"
  type="button"
>
  className
</button>
`;

exports[`<Button /> should render button with disabled 1`] = `
<button
  class="ar-button ar-button--default ar-button-- ar-button--middle ar-button--loading ar-button--disabled"
  type="button"
>
  disabled
</button>
`;

exports[`<Button /> should render button with disabled 2`] = `
<button
  class="ar-button ar-button--default ar-button-- ar-button--middle ar-button--loading"
  type="button"
>
  not disabled
</button>
`;

exports[`<Button /> should render button with loading 1`] = `
<button
  class="ar-button ar-button--default ar-button-- ar-button--middle ar-button--loading"
  type="button"
>
  loading
</button>
`;

exports[`<Button /> should render button with loading 2`] = `
<button
  class="ar-button ar-button--default ar-button-- ar-button--middle"
  type="button"
>
  not loading
</button>
`;

exports[`<Button /> should render button with onClick 1`] = `
<button
  class="ar-button ar-button--default ar-button-- ar-button--middle ar-button--loading"
  type="button"
>
  onClick
</button>
`;

exports[`<Button /> should render button with style 1`] = `
<button
  class="ar-button ar-button--default ar-button-- ar-button--middle ar-button--loading"
  style="color: red;"
  type="button"
>
  style
</button>
`;

exports[`<Button /> should render default 1`] = `
<div>
  <button
    class="ar-button ar-button--default ar-button-- ar-button--middle ar-button--loading"
    type="button"
  >
    default
  </button>
</div>
`;
